src/components/checkbox/checkbox.tsx

```diff
@@ -136,6 +136,14 @@ export class Checkbox {
         this.mdcCheckbox.indeterminate = newValue;
     }
 
+    @Watch('readonly')
+    protected handleReadonlyChange() {
+        this.destroyMDCInstances();
+        requestAnimationFrame(() => {
+            this.initialize();
+        });
+    }
```

Comment:
I'm a little bit wary about the cognetive complexity of the `requestAnimationFrame` and what's the most canonical way of achieving the intent here 😄

I think I might prefer this
```ts
  @Watch('readonly')
  protected handleReadonlyChange() {
    this.destroyMDCInstances();
    this.shouldReinitialize = true;
  }

  componentDidRender() {
    if (this.shouldReinitialize) {
      this.initialize();
      this.shouldReinitialize = false;
    }
  }
```
To me it's more intuitive that the intent is to initialize the MDC component after a render. But I'm open to the fact that the current implementation is the most sound one for all of you. Can I ask your opinion @jgroth?

Reply:
This seems more stable, I'm not sure if it could take more than one frame to render it and if so, the original approach would fail